src/index.js

```diff
@@ -216,16 +255,44 @@ const compare = (condition, value, attributes) => {
 const reduceRule = (rule, attributes) => {
   const result = {};
 
-  for (const [name, condition] of Object.entries(rule)) {
-    const values = getAttributeValues(attributes, name.split('.'));
+  for (let [conditionName, condition] of Object.entries(rule)) {
+    // When a custom attribute is referenced in a rule's key position, we need
+    // to reverse the condition so we can correctly perform in-line value
+    // replacement (replacement only happens when the key position is known).
+    if (conditionName.startsWith(USER_CUSTOM_ATTRIBUTES_PATH)) {
```

Comment:
@swain I had a question for you here:

is knowing the key and not knowing the attribute a hard requirement for reversing?

I chose to always reverse if we see a custom user attribute in the key. It makes the code simpler and does end up with the same results/requirements, correct? Or am I missing something?

Reply:
"always reversing" seems fine, as long as it means the same thing for calls to `enforce`

Reply:
This might actually change when I take off the hard coded `USER_CUSTOM_ATTRIBUTES_PATH`.